src/codemagic/cli/cli_app.py

```diff
@@ -416,7 +421,7 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
-def common_arguments(*class_arguments: Argument):
+def common_arguments(*class_arguments: Argument) -> Callable[[_Fn], _Fn]:
```

Comment:
```suggestion
def common_arguments(*class_arguments: Argument) -> Callable[[Type[CliApp]], Type[CliApp]]:
```

Currently defined return type is not correct. This is a `CliApp` class decorator and should be type-hinted as such.

Reply:
fixed, however i think it still needs to be a generic otherwise the type wouldn't be polymorphic and it would lose the subtypes